sale_order_import/tests/test_sale_order_import.py

```diff
@@ -0,0 +1,231 @@
+            {"name": "Cash", "type": "cash", "code": "CASH"}
+        )
+        self.note = "Test Note"
+
+        # Check the file path and load the test_regular_sale_order.csv
+        self.file_path = os.path.join(
+            "sale_order_import", "tests", "test_regular_sale_order.csv"
+        )
+        self.generated_file = file_open(self.file_path, "rb")
+        self.generated_file = self.generated_file.read()
+
+    def test_01_get_order_dict_flow(self):
+        """
+            This test evaluates the import test sales order CSV
+             file and compares the values with test records.
+        """
+
+        # Note: Here I have called `import_sale_data` method which is the main
+        # method of sale order CSV file. This method covers the
+        # '  get_order_dict' method flow and their values.
+        # And here I am not going to call the method get_order_dict because
+        # this method many arguments and contains the dynamic value
+        # which is not possible to make in a short time. so, for now,
+        # I have called `import_sale_data` method which covers this method and
+        # dynamic values, Here also compares the
+        # values which return by `get_order_dict`.
+        # https://github.com/qrtl/sst-custom/blob/11.0/sale_order_import/wizard/import_sale.py#L240 # noqa
+
+        # Create wizard where the add test_regular_sale_order.csv
+        # and allow to import sale order data.
+        wizard_sale = self.env["import.sale"].create(
+            {
+                "picking_policy": "direct",
+                "customer_invoice_journal_id": self.cash_journal_cash.id,
+                "customer_payment_journal_id": self.cash_journal_cash.id,
+                "asynchronous": False,
+                "input_file": base64.encodestring(self.generated_file),
+                "datas_fname": self.file_path,
+            }
+        )
+
+        # The Main Method called import_sale_data.
+        res = wizard_sale.import_sale_data()
+
+        # `import_sale_data method returns the actions
+        # So here I am taking the domain which
+        # contains the `ids` of error.log object.
+        res_domain = safe_eval(res.get("domain", {}))[0][2]
+
+        # Browse the error.log object ids which return in res_domain
+        error_log = self.env["error.log"].browse(res_domain)
+        sale_order = error_log.sale_order_ids
+        # Sample data from test_regular_sale_order.csv file
+        # Line Product,Line Description,Line Unit Price,Line Qty,Customer,\
+        # Pricelist,Warehouse,Notes,Carrier,Team,Customer Phone/Mobile
+        # ABC,Product A,200,2,,Test Partner,Public Pricelist,\
+        # My Company,Test Note,Free delivery charges,Sales,+1234567890
+
+        # Compares the values with test_regular_sale_order.csv file.
+        self.assertEqual(
+            sale_order[0].partner_id, self.partner_02, "Partner id is incorrect ",
```

Comment:
```suggestion
            sale_order[0].partner_id, self.partner_02
```
Remove the error messages as they are self-explanatory.